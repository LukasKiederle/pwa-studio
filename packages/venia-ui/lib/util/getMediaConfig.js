// TODO: replace with actual graphql calls to the magento instance
// M2 GraphQL doesn't currently support them
export const getMediaConfig = () => {
    return {
        enabled : true,
        links: {
            facebook: 'https://www.facebook.com',
            instagram: 'https://www.instagram.com',
            twitter: 'https://www.twitter.com',
            youtube: 'https://www.youtube.com'
        }
    }
};

