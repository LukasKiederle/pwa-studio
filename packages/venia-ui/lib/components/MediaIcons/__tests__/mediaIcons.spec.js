import React from 'react';
import TestRenderer from 'react-test-renderer';
import MediaIconGroup from '../mediaIconGroup';
import {getMediaConfig} from "../../../util/getMediaConfig";

jest.mock('../../../classify');

const disabledMediaConfig = {
    enabled: false,
    links: {
        facebook: 'https://www.facebook.com',
        instagram: 'https://www.instagram.com',
        twitter: 'https://www.twitter.com',
        youtube: 'https://www.youtube.com'
    }
};

const mediaConfig = getMediaConfig();

test('is enabled', () => {
    expect(mediaConfig.enabled).toBeTruthy();
});

test('is disabled', () => {
    expect(disabledMediaConfig.enabled).toBeFalsy();
});

test('renders correctly a media icon group', () => {
    const component = TestRenderer.create(
        <MediaIconGroup links={mediaConfig.links} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});
