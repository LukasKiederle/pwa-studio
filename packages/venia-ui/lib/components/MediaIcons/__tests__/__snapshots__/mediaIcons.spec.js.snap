// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a media icon group 1`] = `
<div
  className="root"
>
  <a
    className="LogoClass"
    href="https://www.facebook.com"
  >
    <div
      className="root container"
    >
      <img
        alt="Facebook"
        className="image placeholder"
        height={30}
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        title="Facebook"
        width={30}
      />
      <img
        alt="Facebook"
        className="image notLoaded"
        height={30}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="test-file-stub"
        title="Facebook"
        width={30}
      />
    </div>
  </a>
  <a
    className="LogoClass"
    href="https://www.instagram.com"
  >
    <div
      className="root container"
    >
      <img
        alt="Instagram"
        className="image placeholder"
        height={30}
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        title="Instagram"
        width={30}
      />
      <img
        alt="Instagram"
        className="image notLoaded"
        height={30}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="test-file-stub"
        title="Instagram"
        width={30}
      />
    </div>
  </a>
  <a
    className="LogoClass"
    href="https://www.twitter.com"
  >
    <div
      className="root container"
    >
      <img
        alt="Twitter"
        className="image placeholder"
        height={30}
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        title="Twitter"
        width={30}
      />
      <img
        alt="Twitter"
        className="image notLoaded"
        height={30}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="test-file-stub"
        title="Twitter"
        width={30}
      />
    </div>
  </a>
  <a
    className="LogoClass"
    href="https://www.youtube.com"
  >
    <div
      className="root container"
    >
      <img
        alt="Youtube"
        className="image placeholder"
        height={30}
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        title="Youtube"
        width={30}
      />
      <img
        alt="Youtube"
        className="image notLoaded"
        height={30}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="test-file-stub"
        title="Youtube"
        width={30}
      />
    </div>
  </a>
</div>
`;
